nkms/storage/encrypted_file.py

```diff
@@ -1,114 +1,41 @@
-import msgpack
-import os
 import io
-from nacl.utils import random
+import os
+from nkms.storage.header import Header
+from nkms.storage.constants import NONCE_COUNTER_BYTE_SIZE, PADDING_LENGTH
 from nkms.crypto import default_algorithm, symmetric_from_algorithm
 
 
 class EncryptedFile(object):
-    def __init__(self, key, path, mode='rb'):
+    def __init__(self, key, path, header_path):
         """
         Creates an EncryptedFile object that allows the user to encrypt or
         decrypt data into a file defined at `path`.
+
         An EncryptedFile object actually is composed of two files:
         1) The ciphertext -- This is the chunked and encrypted ciphertext
         2) The header -- This contains the metadata of the ciphertext that
             tells us how to decrypt it, or add more data.
 
         :param bytes key: Symmetric key to use for encryption/decryption
-        :param string/bytes path: Path of file to open
-        :param string mode: Mode to use when opening file, default is 'rb'
+        :param bytes path: Path of ciphertext file to open
+        :param bytes header_path: Path of header file
         """
-        self.path = path
-        self.mode = mode
-
         cipher = symmetric_from_algorithm(default_algorithm)
         self.cipher = cipher(key)
 
-    def _build_header(self, version=100, nonce=None, keys=None,
-                      chunk_size=1000000, num_chunks=0, msg_len=0):
-        """
-        Builds a header and returns the msgpack encoded form of it.
-
-        :param int version: Version of the NuCypher header
-        :param bytes nonce: Nonce to write to header, default is random(20)
-        :param list keys: Keys to write to header
-        :param int chunk_size: Size of each chunk in bytes, default is 1MB
-        :param int num_chunks: Number of chunks in ciphertext, default is 0
-        :param int msg_len: Length of the encrypted ciphertext in total
+        # Opens the header file and parses it, if it exists. If not, creates it
+        self.header_path = header_path
+        self.header_obj = Header(self.header_path)
```

Comment:
So, as you can see, you'll need to access some of the header data. Such that: `chunk_size = self.header[b'chunk_size']`.

I'm very much so open to better methods, what do you have in mind? I'd like to see some pseudocode with how you imagine accessing it would look like.

I allow the public access to `self.header` because I imagined that someone would need to see some of the data in the header occasionally and not inside the `EncryptedFile` class alone.